Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/SerializedObjectIDDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,145 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Data;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Infrastructure.ObjectIDStringSerialization;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+using Remotion.Data.DomainObjects.UnitTests.Factories;
+using Remotion.Development.NUnit.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.Parameters;
+
+[TestFixture]
+public class SerializedObjectIDDataParameterDefinitionTest : StandardMappingTest
+{
+  [Test]
+  public void Initialize ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateStorageTypeInformation();
+
+    var parameterDefinition = new SerializedObjectIDDataParameterDefinition(storageTypeInformation);
+
+    Assert.That(parameterDefinition.StorageTypeInformation, Is.SameAs(storageTypeInformation));
+  }
+
+  [Test]
+  public void Initialize_WithIncorrectStorageTypeInformation ()
+  {
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateIntStorageTypeInformation();
+
+    Assert.That(
+        () => new SerializedObjectIDDataParameterDefinition(storageTypeInformation),
+        Throws.InstanceOf<ArgumentException>().With.ArgumentExceptionMessageEqualTo(
+            string.Format("Parameter '{0}' has type '{2}' when type '{1}' was expected.", "storageTypeInformation.DotNetType", typeof(string), typeof(int)),
+            "storageTypeInformation.DotNetType"));
```

Comment:
Please inline the values as much as possible. I like to not use string-format in unit test assertions since they hide errors, e.g. if you flip parameter order. If sensible, using interpolated strings is now an option, e.g. when the type name is to complicated. Here, with the text form being `System.String` and `System.Int32`, I'm not sure even that's warrented.